Upgrade Stylelint & related pacakges

lb- opened the pull request:
- Additional small patch updates to other dev packages

lb- commented:
@laymonage  or @thibaudcolas  - any chance for a review on this. Just a package.json update - no actual code changes as a result

laymonage reviewed (commented):
Could you also please update the pre-commit counterparts? I think they're out of sync and it has been biting me when I make changes to the CSS files. Thanks!

lb- commented:
@laymonage I have just rebased this and updated a few dev deps to their latest (compared to when this PR was created).

I have also moved the eslint deps to dev dependencies.

As for the pre-commit hooks, unless I am missing something, I cannot see this Sphinx project using them.

I can add an issue to request this if you think it's useful for this repo though.

laymonage reviewed (approved):
@lb- Sorry for some reason I thought this was the wagtail repo 🤦 

This looks good to me, but it seems the visual regression tests are failing? I can't see the logs anymore though

lb- commented:
Yeah. The visual regression tests have been failing for a while. I'll raise an issue for that. It's unrelated to this PR
